File: bppython/output.py
import sys
import os

from bppython import *


# for output:
# package args into a list then pass to bpbase
def Output(s, *args):
    if(len(args)) > 0:
      bpbase.Output(s, list(args))
    else:
      bpbase.Output(s, [])


def Success(s, *args):
    if(len(args)) > 0:
      bpbase.Success(s, list(args))
    else:
      bpbase.Success(s, [])


def Warning(s, *args):
    if(len(args)) > 0:
      bpbase.Warning(s, list(args))
    else:
      bpbase.Warning(s, [])


def Error(s, *args):
    if(len(args)) > 0:
      bpbase.Error(s, list(args))
    else:
      bpbase.Error(s, [])


def Debug(s, *args):
    if(len(args)) > 0:
      bpbase.Debug(s, list(args))
    else:
      bpbase.Debug(s, [])


def Changed(s, *args):
    if(len(args)) > 0:
      bpbase.Changed(s, list(args))
    else:
      bpbase.Changed(s, [])


def Flush():
    bpbase.Flush()


def Line(char, n = 80):
    return char * n + "\n" 

File: bpmodule/output/Output.cpp
/*! \file
 *
 * \brief Output and printing functionality (source)
 * \author Benjamin Pritchard (ben@bennyp.org)
 */ 


#include "bpmodule/output/Output.hpp"


namespace bpmodule {
namespace output {


// global function
bool ColorEnabled(void) noexcept;


namespace detail {

void Output_(std::ostream & out, OutputType type, const std::string & str)
{
    if(ColorEnabled())
    {
        std::string st = str;
        if(st.size() == 0)
            return;

        // strip out the last newline character. Makes it look ugly
        bool newline = (st.back() == '\n');
        if(newline)
            st = st.substr(0, st.size()-1);

        switch(type)
        {
            case OutputType::Success:
                out << "\033[32m" << st << "\033[0m";
                break;
            case OutputType::Changed:
                out << "\033[91m" << st << "\033[0m";
                break;
            case OutputType::Debug:
                out << "\033[35m" << st << "\033[0m";
                break;
            case OutputType::Warning:
                out << "\033[33m" << st << "\033[0m";
                break;
            case OutputType::Error:
                out << "\033[37;41m" << st << "\033[0m";
                break;
            default:
                out << st;
                break;

        }
        if(newline)
            out << "\n";
    }
    else
        out << str;
}

}


} // close namespace output
} // close namespace bpmodule
